### PR TITLE
Better error messages

### PR DESCRIPTION
**Pull request by mouadino**
Separate when discovery return empty service and when heartbeat fail to make
debugging easier.

### CODE REVIEW

File: lymph/core/rpc.py [hunk 1]
```diff
@@ -155,16 +157,18 @@ def _pick_instance(self, service):
                 continue
             if connection.is_alive():
                 choices.append(instance)
+        if count == 0:
+            raise NotConnected('servide have no instance')
         if not choices:
-            raise NotConnected('Not connected to %s' % service)
+            raise NotConnected('all %d instances connection are dead' % count)
```
Comment (emulbreh) [inline lymph/core/rpc.py]: Let's just include ``len(service)`` in the message

File: lymph/core/rpc.py [hunk 2]
```diff
@@ -155,16 +157,18 @@ def _pick_instance(self, service):
                 continue
             if connection.is_alive():
                 choices.append(instance)
+        if count == 0:
+            raise NotConnected('servide have no instance')
```
Comment (promulo) [inline lymph/core/rpc.py]: 'service has' :)

File: lymph/core/rpc.py (hunk 1 above)
Comment (promulo) [inline lymph/core/rpc.py]: well, since I'm catching typos: 'all %d instance connections' :D

Comment (mouadino) [inline lymph/core/rpc.py]: @emulbreh that should be the same as ``count`` ? it only that we don't have to loop over instances two times.

Comment (emulbreh) [inline lymph/core/rpc.py]: no need for this optimization, I'd rather save four lines.

Comment (mouadino) [inline lymph/core/rpc.py]: why ? it's the same in the end, just this solution is better optimized and w/o sacrificing readability ?

**Comment by mei-li** [comment]
:+1:

**Pull request merged**